js/models.js

```diff
@@ -137,12 +148,66 @@ class User {
     );
   }
 
+
+  async userFavorite(storyId) {
+    const response = await axios({
+      url: `${BASE_URL}/users/${this.username}/favorites/${storyId}`,
+      method: 'POST',
+      data: { token: this.loginToken }
+    })
+    // console.log(response);
+    const favArr = response.data.user.favorites;
+    this.favorites.push(favArr[favArr.length - 1]);
+
+    // console.log(this.favorites);
+  }
+
+  async removeFavorite(storyId) {
+    const response = await axios({
+      url: `${BASE_URL}/users/${this.username}/favorites/${storyId}`,
+      method: 'DELETE',
+      data: { token: this.loginToken }
+    })
+    this.favorites = response.data.user.favorites
+    // console.log(response.data.user.favorites)
+    // console.log(this.favorites)
+  }
+  isFavorite(story) {
+    return this.favorites.some(s => (s.storyId === story.storyId))
```

Comment:
Currently this logic is doing a nested loop (one to loop through the stories and one to loop through the favorites). If `this.favorites` is an object or a set (since we don't need anything but storyIds), we can remove the inner loop and have `isFavorite` operate with an O(1) time complexity.